bbq/compiler/desugar.go

```diff
@@ -915,116 +926,154 @@ func (d *Desugar) inheritedFunctionsWithConditions(compositeType *sema.Composite
 			})
 			inheritedFunctions[name] = funcs
 		}
-	}
+	})
 
 	return inheritedFunctions
 }
 
-func (d *Desugar) inheritedDefaultFunctions(compositeType *sema.CompositeType, decl *ast.CompositeDeclaration) []ast.Declaration {
-	directMembers := compositeType.Members
-	allMembers := compositeType.GetMembers()
+func (d *Desugar) inheritedDefaultFunctions(
+	compositeType sema.ConformingType,
+	existingFunctions map[string]*ast.FunctionDeclaration,
+	pos ast.Position,
+	declRange ast.Range,
+) []ast.Declaration {
 
-	pos := decl.StartPos
+	inheritedDefaultFunctions := make(map[string]struct{})
 
 	inheritedMembers := make([]ast.Declaration, 0)
 
-	for memberName, resolver := range allMembers { // nolint:maprange
-		if directMembers.Contains(memberName) {
-			continue
-		}
-
-		member := resolver.Resolve(
-			d.memoryGauge,
-			memberName,
-			ast.EmptyRange,
-			func(err error) {
-				if err != nil {
-					panic(err)
-				}
-			},
-		)
-
-		// Only interested in functions.
-		// Also filter out built-in functions.
-		if member.DeclarationKind != common.DeclarationKindFunction ||
-			member.Predeclared {
-			continue
-		}
-
-		// Inherited functions are always from interfaces
-		interfaceType := member.ContainerType.(*sema.InterfaceType)
+	for _, conformance := range compositeType.EffectiveInterfaceConformances() {
```

Comment:
The improvement/fix was to iterate through conformances to find the default functions (similar to how conditions are handled).
The previous approach of resolving through member-resolvers is not alway accurate, because it will miss default functions to whom descendant(s) have declared conditions. (member resolver thinks it as a overridden function, although it only defined conditions, not a body)
i.e:
```cadence
struct interface Foo {
    fun test(_ a: Int) {
        // some body
    }
}

struct interface Bar: Foo {

    // Here it is NOT overridden. Just some conditions are defined.
    fun test(_ a: Int) {
        pre {
                // some conditions
        }
    }
}

struct Test: Bar {}
```